server/utils/PluginManager.ts

```diff
@@ -35,100 +35,58 @@ export enum PluginType {
  * Router. Registering an API plugin causes the router to be mounted.
  */
 type PluginValueMap = {
-  [PluginType.API]: Router;
-  [PluginType.AuthProvider]: Router;
-  [PluginType.EmailTemplate]: typeof BaseEmail;
-  [PluginType.Processor]: typeof BaseProcessor;
-  [PluginType.Task]: typeof BaseTask<any>;
-  [PluginType.UnfurlProvider]: UnfurlSignature;
+  [Hook.API]: Router;
+  [Hook.AuthProvider]: { router: Router; id: string };
+  [Hook.EmailTemplate]: typeof BaseEmail;
+  [Hook.Processor]: typeof BaseProcessor;
+  [Hook.Task]: typeof BaseTask<any>;
+  [Hook.UnfurlProvider]: UnfurlSignature;
 };
 
-export type Plugin<T extends PluginType> = {
+export type Plugin<T extends Hook> = {
+  /** Plugin type */
+  type: T;
   /** A unique ID for the plugin */
-  id: string;
+  id?: string;
```

Comment:
I think I would remove `id` from the type now, to avoid confusion with the `id` in auth provider value

Reply:
![Screenshot 2024-03-16 at 19 34 56 1](https://github.com/outline/outline/assets/15694566/4a2f5613-b857-4875-a787-ff066025ebe5)

Phew! Looks like `id` is deeply integrated...

Reply:
(not posed as a problem, now it's just `config.value.id`)

Reply:
![Screenshot 2024-03-16 at 19 39 33](https://github.com/outline/outline/assets/15694566/7f69eeb5-2b5a-4fc1-8790-e9b435bffc94)

`id` then has to be removed from many other places, like ☝️

Reply:
> Phew! Looks like id is deeply integrated...

Actually it's a good job tsc caught this.

> id then has to be removed from many other places, like ☝️

Do it